tests/protocol/test_hostplugin.py

```diff
@@ -403,6 +403,7 @@ def test_validate_http_request(self):
 
         with mock_wire_protocol.create(DATA_FILE) as protocol:
             test_goal_state = protocol.client._goal_state
+            plugin = protocol.client.get_host_plugin()
```

Comment:
Since now get_host_plugin() always fetches the goal state, this initialization was breaking the code below under the scope of the http_request mock.

I may come back (and also ask your help) to update more tests. Similar issues are all over the code.